app.py

```diff
@@ -107,14 +207,132 @@ def do_logout(request):
+                    answer=answer
+                )
+        questions = Question.all()
+        if questions:
+            submissions = Submission.get_all_for_user(user)
+            l = []
+            for q in questions:
+                if q.id not in [s.question_id for s in submissions]:
+                    l.append(q)
+            if l:
+                question = l[randint(0, len(l) - 1)]
+                x, u, y = make_data(question, user)
+                if x != [] and y != []:
+                    prediction = int(round(guess(x, u, y)))
+                else:
+                    prediction = "Not enough data to predict this question. Keep going!"
+                if 'HTTP_X_REQUESTED_WITH' in request.environ and request.method == "POST":
+                    return Response(
+                        body=json.dumps({"text": question.text,
+                                         "qid": question.id,
+                                         "prediction": prediction,
+                                         }), content_type=b'application/json')
+                return {"question": question, "prediction": prediction}
+        return {"question": None, "prediction": None}
+    else:
+        return HTTPFound(request.route_url('home'))
 
 
-@view_config(route_name="faq", renderer='templates/faqpage.jinja2')
+@view_config(route_name="about", renderer='templates/faqpage.jinja2')
 def faq(request):
     return {}
 
 
+def make_data(question, user):
+    """Gets data from the database and parses it for the Guess function"""
+    u, questions = _get_data(user, question)
+    users, questions = _select_users(u, questions)
+    x = _parse_into_matrix(questions, users)
+    y = []
+    for user_ in users:
+        y.append(Submission.get_answer(user_, question))
+    u = [sub.answer for sub in Submission.get_all_for_user(user)]
+    u = [Submission.get_answer(user, q) for q in questions]
+    return x, u, y
+
+
+def _select_users(u, questions):
+    _u = [u.pop()]
+    _q = []
+    while u:
+        item = u.pop()
+        qu = questions.pop()
+        if len(item) >= 10:
+            _u.append(item)
+            _q.append(qu)
+
+    users = _u[0]
+    for item in _u:
+            users = list(
+                set(users) & set(item)
+            )
+    return users, _q[::-1]
+
+
+def _get_data(user, question):
+    """Queries database for all the questions the user has answered
+    then queries for all the users that have answered all those
+    questions plus the question that the user is currently answering"""
+    users = []
+    questions = [Question.get_question_by_id(sub.question_id)
+                 for sub in Submission.get_all_for_user(user)]
+
+    for q in questions:
+        users.append([User.get_by_id(sub.user_id)
+                      for sub in Submission.get_all_for_question(q)])
+
+    users.append([User.get_by_id(sub.user_id)
+                  for sub in Submission.get_all_for_question(question)])
```

Comment:
With some more though provided to the APIs that each method provides, you could simplify this a bit.  

Imagine an API on the `Submission` model that returns a list of the user_ids of users who answered a given question.

Then imagine an API on the `User` model that gets a list of users when given a list of user_ids.

```python
users.append(User.get_list(Submission.get_users_who_answered(question)))
```

There are a number of ways one might simplify this code to be a bit more readable than it is.  At the same time, it might allow you to streamline the queries made, which would speed this function up significantly.  As it stands, you issue a `get_by_id` query to the User table for every submission in both this list and in the list of lists in lines 295-297 above.  That's a shitload of queries.